recipes/security/api-umbrella/readme.md

```diff
@@ -0,0 +1,469 @@
+service backed with an scalable
+[replica set](https://docs.mongodb.com/v3.2/replication/) of MongoDB instances.
+
+All elements will be running in docker containers, defined in docker-compose
+files. Actually, this recipe focuses on the deployment of the API Umbrella
+frontend, reusing the [mongodb replica recipe](../../utils/mongo-replicaset/readme.md)
+as its backend.
+
+At the time being, other services, such as [Elastic Search](https://www.elastic.co/products/elasticsearch)
+for logging api interactions and QoS are not deployed.
+This is mostly due to the fact that API Umbrella supports only obsolete versions
+of Elastic Search (i.e. version 2, while current version is 6).
+
+The final deployment is represented by the following picture:
+
+<img src='http://g.gravizo.com/g?
+  digraph G {
+      rankdir=LR;
+      	compound=true;
+      	node [shape="record" style="filled"];
+      	splines=line;
+      	Client [fillcolor="aliceblue"];
+      	subgraph cluster {
+      		label="Docker Swarm Cluster";
+      		"Load Balancer" [fillcolor="aliceblue"];
+      		subgraph cluster_0 {
+      			label="API Umbrella stack";
+      			APIUmbrella1 [fillcolor="aliceblue"];
+      			APIUmbrella2 [fillcolor="aliceblue"];
+      			APIUmbrella3 [fillcolor="aliceblue"];
+      		}
+      		subgraph cluster_1 {
+      			label="MongoDB Replica Set stack";
+      			Mongo1 [fillcolor="aliceblue"];
+      			Mongo2 [fillcolor="aliceblue"];
+      			Mongo3 [fillcolor="aliceblue"];
+      		}
+      	}
+      	Client -> "Load Balancer" [label="80",lhead=cluster_0];
+      	"Load Balancer" -> {APIUmbrella1,APIUmbrella2,APIUmbrella3};
+      	APIUmbrella1 -> Mongo1 [lhead=cluster_1];
+      	APIUmbrella2 -> Mongo1 [lhead=cluster_1];
+      	APIUmbrella3 -> Mongo1 [lhead=cluster_1];
+      	Mongo1 -> {Mongo2, Mongo3} [dir="both"];
+  }
+'>
+
+## Prerequisites
+
+Please make sure you read the [welcome page](../../index.md) and followed
+the steps explained in the [installation guide](../../installation.md).
+
+## How to use
+
+Firstly, you need to have a Docker Swarm (docker >= 17.06-ce) already setup.
+If you don't have one, checkout the [tools](../../tools/readme.md) section
+for a quick way to setup a local swarm.
+
+```
+    $ miniswarm start 3
+    $ eval $(docker-machine env ms-manager0)
+```
+
+In case you haven't done it yet for other recipes, deploy `backend` and
+`frontend` networks as described in the
+[installation guide](../../installation.md#creating-the-networks).
+
+API Umbrella needs a mongo database for its backend. If you have already
+deployed Mongo within your cluster and would like to reuse that database, you
+can skip the next step (deploying backend).
+You will just need to pay attention to the variables
+you define for Orion to link to Mongo, namely, `MONGO_SERVICE_URI`
+and `REPLICASET_NAME`. Make sure you have the correct values in `frontend.env`.
```

Comment:
this should be backend.env

Reply:
it's correct frontend. if you already deployed backend, you need only to set values in frontend.